### PR TITLE
Feature plugin unloading

### LINKED ISSUE #4: Maya shut-down fails
While the Wisp renderer can get initialized and destroyed properly via the plug-in, there is an edge case. If the user decides to shut-down Maya without unloading the plug-in, Maya will fail to completely shut-down.

A small window pops up and the editor becomes unresponsive.

Maya does seem to save files just fine, so did not break at all. It is purely the shut-down phase.
maya_shutdown_bug

### PR DESCRIPTION
**Pull request by tntmeijs**
The plug-in would still crash even though "PostQuitMessage(0)" had already been removed in Wisp. These changes will make sure all objects in the plug-in will be deallocated properly as well.

This fixes #4 

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: src/plugin/renderer/renderer.cpp [hunk 1]
```diff
@@ -13,6 +13,9 @@
 #include "d3d12/d3d12_renderer.hpp"
 #include "wisp.hpp"
 
+// Maya API
+#include <maya/M3dView.h>
```
Comment (sainsay) [inline src/plugin/renderer/renderer.cpp]: not needed include

**Review by sainsay — CHANGES_REQUESTED** [review]
one small code change required.
Wisp library is pointing to a different branch than master. please wait/ update to wisp master branch when required pull request has been merged

File: module/wisp/bin/resources/shaders/deferred_composition.hlsl [hunk 2]
```diff
@@ -103,6 +103,9 @@ void main_cs(int3 dispatch_thread_id : SV_DispatchThreadID)
 		//Ao is multiplied with material texture ao, if present
 		ao *= gbuffer_ao;
 
+		//Ao is multiplied with material texture ao, if present
+		ao *= gbuffer_ao;
```
Comment (MeineZ) [inline module/wisp/bin/resources/shaders/deferred_composition.hlsl]: Duplicated multiplication? Probably due to wisp already being updated in the latest plugin version?

File: src/plugin/renderer/material_manager.cpp [hunk 3]
```diff
@@ -17,35 +17,33 @@
...
-	m_mesh_shading_relations.clear();
-	m_surface_shader_shading_relations.clear();
-}
-
 void wmr::MaterialManager::Initialize()
 {
-	LOG("Attempting to get a reference to the renderer.");
-	auto& renderer = dynamic_cast<const ViewportRendererOverride*>(MHWRender::MRenderer::theRenderer()->findRenderOverride(settings::VIEWPORT_OVERRIDE_NAME))->GetRenderer();
-
+	const auto& renderer = dynamic_cast<const ViewportRendererOverride*>(MHWRender::MRenderer::theRenderer()->findRenderOverride(settings::VIEWPORT_OVERRIDE_NAME))->GetRenderer();
```
Comment (MeineZ) [inline src/plugin/renderer/material_manager.cpp:28]: `LOG` is removed intentionally?

Comment (tntmeijs) [reply src/plugin/renderer/material_manager.cpp:28]: Yes.

**Review by MeineZ — APPROVED** [review]
Works fine now! 👍

**Review by sainsay — APPROVED** [review]

**Pull request merged**